app/gui2/src/components/visualizations/TableVisualization.vue

```diff
@@ -158,6 +174,70 @@ const selectableRowLimits = computed(() => {
 })
 const wasAutomaticallyAutosized = ref(false)
 
+const newNodeSelector = computed(() => {
+  let selector
+  switch (config.nodeType) {
+    case COLUMN_NODE_TYPE:
+    case VECTOR_NODE_TYPE:
+      selector = INDEX_FIELD_NAME
+      break
+    case ROW_NODE_TYPE:
+      selector = 'column'
+      break
+    case EXCEL_WORKBOOK_NODE_TYPE:
+      selector = 'Value'
+      break
+    case SQLITE_CONNECTIONS_NODE_TYPE:
+    case POSTGRES_CONNECTIONS_NODE_TYPE:
+      selector = 'Value'
+      break
+  }
+  return selector
+})
+
+const newNodeAction = computed(() => {
+  let identifierAction
+  switch (config.nodeType) {
+    case COLUMN_NODE_TYPE:
+    case VECTOR_NODE_TYPE:
+      identifierAction = 'at'
+      break
+    case ROW_NODE_TYPE:
+      identifierAction = 'at'
+      break
+    case EXCEL_WORKBOOK_NODE_TYPE:
+      identifierAction = 'read'
+      break
+    case SQLITE_CONNECTIONS_NODE_TYPE:
+    case POSTGRES_CONNECTIONS_NODE_TYPE:
+      identifierAction = 'query'
+      break
+  }
+  return identifierAction
+})
+
+const onClickTooltipValue = computed(() => {
+  let val
+  switch (config.nodeType) {
+    case COLUMN_NODE_TYPE:
+    case VECTOR_NODE_TYPE:
+    case ROW_NODE_TYPE:
+      val = 'value'
+      break
+    case EXCEL_WORKBOOK_NODE_TYPE:
+      val = 'sheet'
+      break
+    case SQLITE_CONNECTIONS_NODE_TYPE:
+    case POSTGRES_CONNECTIONS_NODE_TYPE:
+      val = 'table'
+      break
+    case TABLE_NODE_TYPE:
+    case DB_TABLE_NODE_TYPE:
+      val = 'row'
+  }
+  return val
+})
```

Comment:
The logic might be easier to follow with one computed/switch selecting between different constant objects, with each object having fields like `selector`/`identifierAction`/`tooltipValue`. Up to you.

Reply:
+1 I wanted to write the same thing